cli/src/commands/snyk/snyk.rs

```diff
@@ -0,0 +1,208 @@
+use std::collections::HashMap;
+use std::{fs, env};
+use async_trait::async_trait;
+use serde_json::Value;
+use crate::commands::Provider;
+use crate::commands::snyk::snyk_data_model::{SnykData, ProjectJson, ProjectDetails};
+use platform::hyper::{ContentType, Error as HyperError, get, post};
+use platform::auth::get_secret;
+use harbor_client::client::simple_upload_sbom;
+use anyhow::Result;
+//DONE: Add method call
+//DONE: See if it runs
+//DONE: Pull in HTTP library
+//DONE: Get org count from snyk repo
+//DONE: Get projects for each org
+//DONE: Get SBOMS for each project
+//DONE: Save SBOMS to temp location
+//DONE: Restructure and cleanup pass 1
+//DONE: Merge Quinns code
+//DONE: Restructure and cleanup pass 2
+//DONE: Rebase main
+//TODO: Event logging
+//TODO: Send SBOM somewhere and review solution works...
+//TODO: Documentation
+//TODO: Final Restructure and cleanup pass
+
+pub struct SnykProvider {}
+
+impl SnykProvider {
+
+    //Origins that can have associated SBOM data in Snyk
+    const VALID_ORIGINS: &'static [&'static str] = &["cli", "github", "github-enterprise", "gitlab"];
+    //Types that can have associated SBOM data in Snyk
+    const VALID_TYPES: &'static [&'static str] = &["npm", "nuget", "gradle", "hex", "pip", "poetry", "rubygems", 
+    "maven", "yarn", "yarn-workspace", "composer", "gomodules", "govendor", "golang", "golangdep", "gradle", "paket", "cocoapods", "cpp", "sbt"];
+    //URL used for getting org and project details
+    const API_V1_URL: &'static str = "https://snyk.io/api/v1";
+    //URL used for getting SBOMS
+    const API_V3_URL: &'static str = "https://api.snyk.io/rest";
+    const API_V3_VERSION: &'static str = "2023-02-15~beta";
+    //Format of SBOMs returned from Snyk
+    const SBOM_FORMAT: &'static str = "cyclonedx%2Bjson";
+    const AWS_SECRET_NAME: &'static str = "dev-harbor-snyk-token-use1";
+
+    //TODO: delete this
+    const TEMP_TEST_DIR: &'static str = "/home/jshattjr";
+
+    // Retrieves a list of all Orgs from Snyk and adds them into a new SnykData object
+    async fn get_orgs(snyk_token: &str) -> SnykData{
+        println!("Getting list of Orgs from Snyk...");
+        let url = format!("{}/orgs", Self::API_V1_URL);
+
+        let response: Result<Option<SnykData>, HyperError> = get(
+            url.as_str(),
+            ContentType::Json,
+            snyk_token,
+            None::<String>,
+        ).await;
+    
+        match response {
+            Ok(option) => 
+                match option {
+                Some(value) => {return value},
+                None => panic!("empty data"),
```

Comment:
Yeah I was wondering how best to do this, I left the panics in as they are better than nothing, but I agree its bad form to kill the whole application on any error. I am curious on how you would suggest I go about this? I was thinking it might be useful to make a Snyk Error class that can be used when something bad happens.

Reply:
We have an `Error` enum!  Check it out [here](https://github.com/CMS-Enterprise/sbom-harbor/blob/main/core/src/errors.rs).   It uses [thiserror](https://crates.io/crates/thiserror) in case you want to read up on the docs and see more examples.